sdk/nodejs/runtime/closure/createClosure.ts

```diff
@@ -1220,23 +1317,76 @@ function findNormalizedModuleName(obj: any): string | undefined {
     return undefined;
 }
 
-const legalNameRegex = /^[a-zA-Z_][0-9a-zA-Z_]*$/;
-export function isLegalMemberName(n: string) {
-    return legalNameRegex.test(n);
+async function hasTruthyMemberAsync(obj: any, memberName: string): Promise<boolean> {
+    if (!obj) {
+        return false;
+    }
+
+    return obj[memberName] ? true : false;
 }
 
-export function isLegalFunctionName(n: string) {
-    if (!isLegalMemberName(n)) {
-        return false;
+function createClosurePropertyDescriptor(
+    nameOrSymbol: string | symbol, descriptor: PropertyDescriptor): ClosurePropertyDescriptor {
+
+    if (nameOrSymbol === undefined) {
```

Comment:
this should be a tautology if our types are correct, right? are we missing an ` | undefined` somewhere?

Reply:
*if* our types are correct.  But we're effectively doing reflection.  so i'm being paranoid.